### ISSUE TITLE
Alert “Unable to create new file from template”

### ISSUE THREAD

**Issue by chenweiwen07** [nextcloud/server#26672]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Open the left menu "Favorites".（左侧菜单“收藏”）
2. Right click any file and click "view in folder".(右键任意文件，点击“在文件夹中查看”)
3. Click "New text document" in current position.(当前位置点击“新建文本文档”)

###  图片展示
![1618974862354](https://user-images.githubusercontent.com/35797132/115491569-c93bda80-a292-11eb-9a8b-9a6644975f5c.jpg)

### Expected behaviour
New text document.（新建文本文档）

### Actual behaviour
Alert “Unable to create new file from template”.（提示“无法从模板创建新文件”）

### 启动脚本
```
#!/bin/bash

set -ex

VERSION=21.0.1

docker pull nextcloud:$VERSION
docker rm -f my-nextcloud
docker run -d --name my-nextcloud \
  -v /home/ubuntu/MyDisk2/nextcloud:/var/www/html \
  --restart always \
  -p 57202:80 \
  nextcloud:$VERSION
docker cp sources.list my-nextcloud:/etc/apt/sources.list
docker exec -it my-nextcloud bash -c "apt update && apt install -y smbclient libmagickcore-6.q16-6-extra"
docker image prune -a -f
```

### Server configuration

**Operating system: ubuntu**

**Web server:**

**Database:**

**PHP version:**

**Nextcloud version: 21.0.1** (see Nextcloud admin page)

**Updated from an older Nextcloud/ownCloud or fresh install:**

**Where did you install Nextcloud from: docker hub**

###

**Comment by wiswedel** [comment]
@chenweiwen07 Could you please re-state your issue in English? Thx.

**Comment by chenweiwen07** [comment]
> @chenweiwen07 Could you please re-state your issue in English? Thx.

I'll try.You can take a look.

**Comment by chenweiwen07** [comment]
@wiswedel I'll try.You can take a look.

**Comment by szaimen** [comment]
Hi @chenweiwen07, can you please update to NC21.0.2 and try again? I know that there were some improvements to this part of the code. Maybe it is already fixed with the new version.

**Comment by chenweiwen07** [comment]
> Hi @chenweiwen07, can you please update to NC21.0.2 and try again? I know that there were some improvements to this part of the code. Maybe it is already fixed with the new version.

I can't find nc21.0.2 of arm64 in docker hub. I upgraded to latest, but it's still 21.0.1.

**Comment by szaimen** [comment]
So then please report back after you've upgraded to NC21.0.2, thanks! :)

**Comment by chenweiwen07** [comment]
> So then please report back after you've upgraded to NC21.0.2, thanks! :)

I upgraded to NC21.0.2, but the bug has not been fixed.

**Comment by szaimen** [comment]
Thanks! So then cc @juliushaertl

**Comment by juliushaertl** [comment]
Anything in the Nextcloud log file related to the failure?

**Comment by chenweiwen07** [comment]
> Anything in the Nextcloud log file related to the failure?

On the demo website([https://try.nextcloud.com/](https://try.nextcloud.com/)),this problem is easy to reproduce.

**Comment by chenweiwen07** [comment]
> > Anything in the Nextcloud log file related to the failure?
> 
> On the demo website(https://try.nextcloud.com/),this problem is easy to reproduce.

@juliushaertl

**Comment by Jesse-Ma** [comment]
你的那些文件是不是用命令 scan 出来的.   看一下数据的文件夹及其子文件夹的owner 是不是有问题.  一般都是www-data

**Comment by chenweiwen07** [comment]
* 首先，在官网体验网页[https://try.nextcloud.com/](https://try.nextcloud.com/)也可以重现该问题
* 其次，我的文件权限如下
![image](https://user-images.githubusercontent.com/35797132/131214523-d59c6a7d-d9bf-4389-a408-2efe968030c1.png)
* 最后，不明白你说的scan命令是指什么，我只是正常通过页面使用nextcloud
* 问题所在，仅通过上述操作，进入相应目录，创建文件时通过浏览器debug可看出，参数里没有正确路径
![image](https://user-images.githubusercontent.com/35797132/131214767-c15c2029-55a7-409e-b22b-5fd50bd2e184.png)
@Jesse-Ma

**Comment by slater0013** [comment]
Hello,

Nextcloud 21.0.4 hosted on Debian 10.

Same bug here but not 100% reproductible ! Sometimes the file creation process goes well.

The difference is in the POST payload, the filePath is sometimes incorrect : 

```
{
    "filePath": "/null/dguherz.md"
}
```

Tested on Chrome & Firefox (last versions) on Windows 10 et Mac OS X.

Regards,
GS.

**Comment by slater0013** [comment]
> Hello,
> 
> Nextcloud 21.0.4 hosted on Debian 10.
> 
> Same bug here but not 100% reproductible ! Sometimes the file creation process goes well.
> 
> The difference is in the POST payload, the filePath is sometimes incorrect :
> 
> ```
> {
>     "filePath": "/null/dguherz.md"
> }
> ```
> 
> Tested on Chrome & Firefox (last versions) on Windows 10 et Mac OS X.
> 
> Regards,
> GS.

Ok, so when accessing the folder directly from the homepage, It's ok !

But the error appears when accessing the folder via "Shared with others" or "Shared with me".

Same thin for accessing the folder via "Favorites".

GS.

**Comment by mvvvmd** [comment]
Same issue when creating Office files with Onlyoffice. If you have templates it fails creation, if you don't it does create the file.

**Comment by skjnldsv** [comment]
Hello everyone!
I found one bug in the template process, but I'm not sure that is also the issue here.
Can anyone that have a reproductible example screenshot the entire expanded stacktrace from the console?
e.g.:
![image](https://user-images.githubusercontent.com/14975046/133746530-672f77d0-7632-4218-a898-9da320af4721.png)

**Comment by mvvvmd** [comment]
Attached the log, it was to long for a screenshot. This is with creating a new OnlyOffice presentation when I have a Templates folder. I have the issue when creating other docs but this is not consistent.

[nc_template_error-1631865855668.log](https://github.com/nextcloud/server/files/7183867/nc_template_error-1631865855668.log)

**Comment by skjnldsv** [comment]
@mvvvmd thanks! That is a different issue than what op reported :)
Can you open a dedicated one in the onlyoffice repo (if it doesn't already exists?)
![image](https://user-images.githubusercontent.com/14975046/133749250-5a1112a9-f8d9-4a7a-8321-10a874cd2b4f.png)

**Comment by mvvvmd** [comment]
> @mvvvmd thanks! That is a different issue than what op reported :)
> Can you open a dedicated one in the onlyoffice repo (if it doesn't already exists?)

I see, I will do that, thank you.

**Comment by skjnldsv** [comment]
> On the demo website([try.nextcloud.com](https://try.nextcloud.com/)),this problem is easy to reproduce.

I can't :shrug:

**Comment by skjnldsv** [comment]
Potential fix :https://github.com/nextcloud/server/pull/28871

**Comment by putt1ck** [comment]
Noting issue has been closed but problem persists into NC 22.2.3 for new text docs; looks like some sort of initialisation issue, because a fix was accidentally discovered - as the instance also had drawio installed, a quick check found that creating new diagrams worked - and after creating a new diagram, creating a new text document also works. 

The workaround seemed to be a permanent (so far!) fix for folders accessed via the files app "home" but it was noted it has to be reapplied for folders accessed through "shared with others" etc. - which lead to discovery that it has to be reapplied to folders that are shared (whether from, to or by link).

This is a live install that updates on the beta channel, initially built a while ago from source on then Ubuntu 16.04, now running on 20.04.

**Comment by szaimen** [comment]
Hi @putt1ck thanks for your feedback! Could you please open a new issue with your findings and post your server config and other needed info there? Thank you!

**Comment by bitshiftnetau** [comment]
Still not fixed. This issue should not be closed.

**Comment by NiTRoeSE** [comment]
We have the same problem in combination with onlyoffice .. sometimes it works, most of the time it doenst.

`{file=/var/www/nextcloud/apps/files/lib/Controller/TemplateController.php, line=57, function=createFromTemplate, class=OC\Files\Template\TemplateManager, type=->, args=[/null/vbv.md, , user]}, {file=/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php, line=217, function=create, class=OCA\Files\Controller\TemplateController, type=->, args=[/null/vbv.md, , user]}, {file=/var/www/nextcloud/lib/private/AppFramework/Http/Dispatcher.php, line=126, function=executeController, class=OC\AppFramework\Http\Dispatcher, type=->, args=[{__class__=OCA\Files\Controller\TemplateController}, create]}, {file=/var/www/nextcloud/lib/private/AppFramework/App.php, line=156, function=dispatch, class=OC\AppFramework\Http\Dispatcher, type=->, args=[{__class__=OCA\Files\Controller\TemplateController}, create]}, {file=/var/www/nextcloud/lib/private/Route/Router.php, line=302, function=main, class=OC\AppFramework\App, type=::, args=[OCA\Files\Controller\TemplateController, create, {__class__=OC\AppFramework\DependencyInjection\DIContainer}, {_route=ocs.files.Template.create}]}, {file=/var/www/nextcloud/ocs/v1.php, line=62, function=match, class=OC\Route\Router, type=->, args=[/ocsapp/apps/files/api/v1/templates/create]}, {file=/var/www/nextcloud/ocs/v2.php, line=23, args=[/var/www/nextcloud/ocs/v1.php], function=require_once}`

Any help, workaround or fix would be awesome.
Thanks

**Comment by aptiko** [comment]
We still have this problem in a new 23.0.0 installation.

**Comment by zefzefzefzef** [comment]
had same issue. creating a template folder via the menu fixed it for me

**Comment by tj0xin** [comment]
Having the same problem with Latest at the time of writing.

**Comment by baselab** [comment]
Same issue here via "nextcloud:latest"
Fixed creating a new directory next to Templates and the other dirs (name is irrelevant and it can be removed afterwards)
Thanks [zefzefzefzef](https://github.com/zefzefzefzef)

**Comment by zachrybaker** [comment]
Issue remains with latest.  Should be reopened.
In my case in the console I see 

````
typeError: OCA.Files.fileActions.getDefaultFileAction(...) is undefined
    e TemplatePicker.vue:222
    c runtime.js:63
    _invoke runtime.js:294
    g runtime.js:119
    $ main.js:358
    i main.js:358
    promise callback*$ main.js:358
    i main.js:358
    promise callback*$ main.js:358
    i main.js:358
    M main.js:358
    M main.js:358
    onSubmit TemplatePicker.vue:197
    submit TemplatePicker.vue:1
````

**Comment by biva** [comment]
Hello, we have the same issue (NC 23 + OnlyOffice 7.0.1) @szaimen , would it be possible to reopen this issue? Or to explain how to fix it? I'm sorry but I didn't understand the workarounds mentioned by @zefzefzefzef and @putt1ck : would anybody be able to give some more detail? Is there another open issue that we should follow?

Thanks for your support!

**Comment by drhirn** [comment]
Same.
Nextcloud Hub II 23.0.4

Console Output:
```
[ERROR] files: Error while creating the new file from template 
Object { app: "files", uid: "<uid>" }
ConsoleLogger.js:41:16
    log ConsoleLogger.js:41
    error ConsoleLogger.js:63
    e TemplatePicker.vue:233
    c runtime.js:63
    _invoke runtime.js:294
    g runtime.js:119
    $ main.js:358
    i main.js:358
    (Async: promise callback)
    $ main.js:358
    i main.js:358
    (Async: promise callback)
    $ main.js:358
    i main.js:358
    M main.js:358
    M main.js:358
    onSubmit TemplatePicker.vue:197
    r TemplatePicker.vue:170
    c runtime.js:63
    _invoke runtime.js:294
    g runtime.js:119
    $ main.js:358
    i main.js:358
    (Async: promise callback)
    $ main.js:358
    i main.js:358
    M main.js:358
    M main.js:358
    open TemplatePicker.vue:155
    actionHandler templates.js:113
    _promptFileName merged-index.js:11541
    jQuery 8
    _promptFileName merged-index.js:11518
    jQuery 10
    _promptFileName merged-index.js:11515
    _onClickAction merged-index.js:11446
    jQuery 2
```
and
```
TypeError: OCA.Files.fileActions.getDefaultFileAction(...) is undefined
    e TemplatePicker.vue:222
    c runtime.js:63
    _invoke runtime.js:294
    g runtime.js:119
    $ main.js:358
    i main.js:358
    promise callback*$ main.js:358
    i main.js:358
    promise callback*$ main.js:358
    i main.js:358
    M main.js:358
    M main.js:358
    onSubmit TemplatePicker.vue:197
    r TemplatePicker.vue:170
    c runtime.js:63
    _invoke runtime.js:294
    g runtime.js:119
    $ main.js:358
    i main.js:358
    promise callback*$ main.js:358
    i main.js:358
    M main.js:358
    M main.js:358
    open TemplatePicker.vue:155
    actionHandler templates.js:113
    _promptFileName merged-index.js:11541
    jQuery 8
    _promptFileName merged-index.js:11518
    jQuery 10
    _promptFileName merged-index.js:11515
    _onClickAction merged-index.js:11446
    jQuery 2
TemplatePicker.vue:234
    e TemplatePicker.vue:234
    c runtime.js:63
    _invoke runtime.js:294
    g runtime.js:119
    $ main.js:358
    i main.js:358
    (Async: promise callback)
    $ main.js:358
    i main.js:358
    (Async: promise callback)
    $ main.js:358
    i main.js:358
    M main.js:358
    M main.js:358
    onSubmit TemplatePicker.vue:197
    r TemplatePicker.vue:170
    c runtime.js:63
    _invoke runtime.js:294
    g runtime.js:119
    $ main.js:358
    i main.js:358
    (Async: promise callback)
    $ main.js:358
    i main.js:358
    M main.js:358
    M main.js:358
    open TemplatePicker.vue:155
    actionHandler templates.js:113
    _promptFileName merged-index.js:11541
    jQuery 8
    _promptFileName merged-index.js:11518
    jQuery 10
    _promptFileName merged-index.js:11515
    _onClickAction merged-index.js:11446
    jQuery 2
```

**Comment by lokedhs** [comment]
> Hi @putt1ck thanks for your feedback! Could you please open a new issue with your findings and post your server config and other needed info there? Thank you!

I'm adding this note for information in case someone is looking into this. I saw this problem today after enabling full text search. Your workaround worked.

**Comment by lokedhs** [comment]
> 



> Noting issue has been closed but problem persists into NC 22.2.3 for new text docs; looks like some sort of initialisation issue, because a fix was accidentally discovered - as the instance also had drawio installed, a quick check found that creating new diagrams worked - and after creating a new diagram, creating a new text document also works.

I'm adding this note for information in case someone is looking into this. I saw this problem today after enabling full text search. Your workaround worked.
